src/gt4py/next/otf/binding/pybind.py

```diff
@@ -146,18 +155,36 @@ def visit_BufferSID(self, sid: BufferSID):
             )"""
     )
 
+    def visit_CompositeSID(self, node: CompositeSID, **kwargs):
+        kwargs["composite_ids"] = (
+            f"gridtools::integral_constant<int,{i}>" for i in range(len(node.elems))
+        )
+        return self.generic_visit(node, **kwargs)
+
+    CompositeSID = as_jinja(
+        "gridtools::sid::composite::keys<{{','.join(composite_ids)}}>::make_values({{','.join(elems)}})"
+    )
+
     DimensionType = as_jinja("""generated::{{name}}_t""")
 
 
-def make_argument(param: interface.Parameter) -> str | BufferSID:
-    if isinstance(param.type_, ts.FieldType):
+def _tuple_get(index: int, var: str) -> str:
+    return f"gridtools::tuple_util::get<{index}>({var})"
+
+
+def make_argument(name: str, type_: ts.TypeSpec) -> str | BufferSID | CompositeSID:
+    if isinstance(type_, ts.FieldType):
         return BufferSID(
-            source_buffer=param.name,
-            dimensions=[DimensionType(name=dim.value) for dim in param.type_.dims],
-            scalar_type=param.type_.dtype,
+            source_buffer=name,
+            dimensions=[DimensionType(name=dim.value) for dim in type_.dims],
+            scalar_type=type_.dtype,
+        )
+    if isinstance(type_, ts.TupleType):
```

Comment:
I would also restrict this to tuples of fields (`primitive_constituents`).

Reply:
done and added a NotImplementedError for the scalar case.